pkg/types/coin_test.go

```diff
@@ -0,0 +1,351 @@
+package types_test
+
+import (
+	"crypto/sha256"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/chia-network/go-chia-libs/pkg/types"
+)
+
+var (
+	parentCoin  types.Bytes32
+	puzzleHash  types.Bytes32
+	puzzleHash2 types.Bytes32
+)
+
+func init() {
+	var err error
+
+	parentCoin, err = types.BytesToBytes32([]byte("---foo---                       "))
+	if err != nil { panic(err) }
+
+	puzzleHash, err = types.BytesToBytes32([]byte("---bar---                       "))
+	if err != nil { panic(err) }
+
+	puzzleHash2, err = types.BytesToBytes32([]byte("---bar--- 2                     "))
+	if err != nil { panic(err) }
+}
+
+// TestCoinID coin id tests adapted from https://github.com/Chia-Network/chia_rs/blob/main/tests/test_coin.py
+// With additional static/expected hex values of the coin ID, just for my own sanity
+// Static values computed using python like: Coin(parent_coin, puzzle_hash, 0xFF).name().hex()
+func TestCoinID(t *testing.T) {
```

Comment:
I tend to prefer smaller test functions like `TestCoinID_FFPrefix` or something like that but one 320 line test function works too :P

Reply:
lol, yes. This test is also basically a port of the rust test, so kept it structured the same.